### ISSUE TITLE
Pasting some long text into the input can cause the entire long text to be highlighted

### ISSUE THREAD

**Issue by fullStackOasis** [dabakovich/react-native-controlled-mentions#118]
The bug is happening in a complex project that I'm working on, but I am able to reproduce this using the example project in the source code here.

I cloned your project, and made a few small changes, but the changes were for clarity. You can demo the issue with your own example code and no changes.

Steps to reproduce:

Copy into your clipboard the following text (select/copy): `is a simple python script that uses yt-dlp to scrape all of a youtube channels`
Run your example project.
Press into the input and delete the default text.
Press on the input, and type the "@" character, which brings up the user suggestions. Choose David Tabaka.
Type a single space character. You do not need to type the space character if you've set `isInsertSpaceAfterMention: true` for the "@" trigger because it is automatically inserted.
Long press in the input to get the Android menu with various options. Choose the "paste as plain text" option.
The text that you copied is pasted into the input, and almost immediately you see the entire content displayed as part of the controlled mention.

You can checkout my project, install it, and run it to demo the issue.
https://github.com/fullStackOasis/react-native-controlled-mentions-example

Once the bug happens, I've found that the app eventually crashes if I continue typing enough. As I type, all the text shows up as a "mention".

The bug only happens for some text that I copied. With the text that I wrote above, it is 100% reproducible for me.

Any support you can offer is much appreciated!

**Comment by fullStackOasis** [comment]
Screenshot showing the issue:

![image](https://github.com/dabakovich/react-native-controlled-mentions/assets/59945095/90c5e49f-25a9-4f68-9d97-012539f7b141)

**Comment by fullStackOasis** [comment]
It appears that this is a React Native bug with `TextInput` and some specific Android devices. I opened an issue here: https://github.com/facebook/react-native/issues/37502

**Comment by dabakovich** [comment]
Hey @fullStackOasis, thank you for such a clear and detailed feedback!

Sorry for the long wait, I'm currently focused on other projects, and the support for this library is not as active as it used to be.

Please let me know, is this question still relevant? If so, have you tried using the prerelease v3 version of this library? I tried to reproduce your issue on my end with the 3rd version, and the problem didn't occur.

https://github.com/dabakovich/react-native-controlled-mentions/releases/tag/v3.0.0-alpha.2

**Comment by fullStackOasis** [comment]
@dabakovich Thank you for getting back to this.

I tried installing [v3.0.0-alpha.4](https://github.com/dabakovich/react-native-controlled-mentions/releases/tag/v3.0.0-alpha.4) - that was the latest - and made a few changes to [my sample project](https://github.com/fullStackOasis/react-native-controlled-mentions-example/tree/v3.0.0-alpha.2), but the problem is still there.

I'm not sure that this problem is in your project, in fact. There's already a bug in React Native in `TextInput` on some specific devices, so it could be that this is the true problem.

**Comment by fullStackOasis** [comment]
@dabakovich Sorry to say that this is still an issue. I just grabbed your demo code and updated my sample project so that I could use your v3 of this package. The app runs, but the bug persists. I've attached a screen recording in [my sample project](https://github.com/fullStackOasis/react-native-controlled-mentions-example) that shows the bug still. This is with commit [d3c6877](https://github.com/fullStackOasis/react-native-controlled-mentions-example/commit/d3c68771fc24ffcecfbd0d808bb2133cdfa736a8)


I'll try to install Google Keyboard, and let you know what happens.

**Comment by dabakovich** [comment]
@fullStackOasis thank you so much for the in-depth investigation and [detailed description](https://github.com/fullStackOasis/react-native-controlled-mentions-example/tree/main?tab=readme-ov-file#june-1-2025-testing-gboard)!

That's good that you found a workaround for the issue. But potential forcing end Samsung users to use GBoard definitely not the best solution.

Seems like iOS keyboards with cjk (chinese, japanese, korean) languages have similar issue (#121). However, in their case, `isInsertSpaceAfterMention` "fixes" the issue. One more thing that you can try to play with is using zero-width space (`\u200B`), thank could be inserted after a trigger part in the `generateValueWithAddedSuggestion`.

Anyway, please, feel free to re-open the ticket or leave comments once you'll get any new insights/ideas/workaround/issues.

**Comment by fullStackOasis** [comment]
I agree installing Gboard is not a good solution @dabakovich and I might try to bring this issue to the attention of Samsung if I get a little more time.
That's interesting about #121 - might give a clue to a different workaround. I'll reopen if I make any progress.
